### DISCUSSION TITLE
Hybrid approach?  Build + Runtime approach

### DISCUSSION THREAD

**Discussion by paulm17** [chakra-ui/panda#719] [Ideas]
PandaCSS is the leading css-in-js library in terms of features.  

However, I would really love to see a hybrid approach for dynamic variables.

Something that is akin to this:  https://github.com/poteboy/kuma-ui/issues/53

Maybe have a runtime utility similar to that of twind?

**Comment by segunadebayo** [comment]
We don't have it in mind to create this for now.

The hybrid approach has challenges that get more complex with SSR and RSC.

I thought about allowing users to opt-in to a "runtime" mode that integrates with existing tools like Styled Components, Emotion, etc. We stopped working on this to reduce the scope of the project and avoid a large API surface area.
